lua/includes/modules/callable_properties.lua

```diff
@@ -0,0 +1,73 @@
+CallableProperties = {}
+
+local function setup(object)
+    local meta = getmetatable(object)
+
+    -- Handles situations where the metatabale's __index was already set
+    local originalIndex
+    if meta then
+        originalIndex = meta.__index
+    end
+
+    setmetatable(object, {
+        __callableProperties = {},
+        __originalIndex = originalIndex,
+        __index = function (self, index)
+            local meta = getmetatable(self)
+
+            local storedValue = meta.__callableProperties[index]
+            if storedValue then
+                if storedValue == nil then return end
+                if type(storedValue) ~= "function" then return storedValue end
+
+                return storedValue(object)
+            end
+
+            -- Super gross way to stop infinite recursion
+            local copyobj = table.Copy(object)
+            local copyMeta = table.Copy(getmetatable(copyobj))
```

Comment:
The issue here is that if we try to fallback to the default "just get the value from the table" and the given `index` doesn't exist in the table, it'll actually call our custom `__index` function again, which get stuck in an infinite loop.

Reply:
How do you see that happening?

Reply:
How i'd implement the change, or how problems will occur?